experimental/dds/tree2/src/feature-libraries/treeTextCursor.ts

```diff
@@ -76,3 +78,25 @@ export function jsonableTreeFromCursor(cursor: ITreeCursor): JsonableTree {
 	}
 	return node;
 }
+
+/**
+ * Extract a JsonableTree from the contents of the given ITreeCursor's current node.
+ */
+export function jsonableTreeFromFieldCursor(cursor: ITreeCursor): JsonableTree[] {
+	assert(cursor.mode === CursorLocationType.Fields, "must start at field");
+	return mapCursorField(cursor, jsonableTreeFromCursor);
+}
+
+/**
+ * Copy forest content into a JsonableTree.
+ * @remarks
+ * This is not a time or memory efficient way to pass around forest content:
+ * its intended for debugging and testing purposes when forest content is needed in a human readable serializable format.
+ */
+export function jsonableTreeFromForest(forest: IForestSubscription): JsonableTree[] {
+	const readCursor = forest.allocateCursor();
```

Comment:
any reason you try-finally'd the cursor allocation in `contentSnapshot` but not here?

In general I think we're a bit inconsistent in this realm.

Reply:
That was not done for exception safety: the one I did a try finally on contained a line with a return that referenced the cursor. I simply choose to try finally it instead of declaring a local and splitting up the return.

Ideally we would be using the new `using` feature from TS 5.2, but thats not compatible with this API, and I don't think we can actually use it yet.